File: Kurs C++/Zad6/modulo.cpp
//
// Created by Mateusz Pater on 08.04.2016.
//

#include "modulo.h"



double modulo::oblicz() {
    return fmod(a->oblicz(), b->oblicz());
}

modulo::modulo(wyrazenie *a, wyrazenie *b) {
    this->a = a;
    this->b = b;
    if( b->oblicz() == 0 ) {  throw std::invalid_argument("Dzielenie przez 0"); }
}

std::string modulo::formatuj_wyrazenie(){
    return a->formatuj_wyrazenie() + "%" + b->formatuj_wyrazenie();
}

File: Kurs Java/Zadanie 3/src/com/company/Modulo.java
/*
 * Copyright (c) 2016. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
 * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
 * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
 * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
 * Vestibulum commodo. Ut rhoncus gravida arcu.
 */

package com.company;

/**
 * author @pater
 */
public class Modulo extends Operator2Arg {

    public Modulo(Wyrazenie a, Wyrazenie b) {
        super(a,b);

    }

    @Override
    public double oblicz() {

        if(b.oblicz() == 0)
        {
            throw new IllegalArgumentException("Cholero, nie dziel przez 0:>");
        }

        return a.oblicz() % b.oblicz();
    }
    @Override
    public String toString() {
        return  a + "mod" + b;
    }
}
